### ISSUE TITLE
Intermittent failure in TestCanceledRefresh

### ISSUE THREAD

**Issue by ellismg** [pulumi/pulumi#2285]
I've seen this in at least two CI legs over the past week or so.

```
--- FAIL: TestCanceledRefresh (0.00s)
	Error Trace:	lifecycle_test.go:1796
	Error:		An error is expected but got nil.
        
    lifecycle_test.go:1798: 3/3 resources refreshed
FAIL
```

I have no idea how to reproduce this or where the issue might be.

**Comment by CyrusNajmabadi** [comment]
Hit this again in: https://travis-ci.com/pulumi/pulumi/jobs/166283425

**Comment by lukehoban** [comment]
I've seen this running tests locally as well.

**Comment by pgavlin** [comment]
Closed with #2331.
